tests/test_cli_options.py

```diff
@@ -0,0 +1,507 @@
+        ]
+        #yapf: enable
+
+        all_tested_options_set = set()
+
+        for option in options:
+            all_tested_options_set.add(self._convert_flag(option.long_flag))
+
+            if option.type in ["bool"]:
+                self._test_boolean_option(option)
+            elif option.type in ["int", "float"]:
+                self._test_numeric_option(option)
+            elif option.type in ["string"]:
+                self._test_string_option(option)
+            elif option.type in ["intlist", "stringlist"]:
+                self._test_list_option(option)
+
+        self._verify_all_options_tested(all_tested_options_set)
+
+    def _verify_all_options_tested(self, all_tested_options_set):
+        cli_option_set = set()
+
+        # Get all of the options in the CLI Configs
+        structs = [
+            CLIConfigProfileStruct, CLIConfigAnalyzeStruct,
+            CLIConfigReportStruct
+        ]
+        for struct in structs:
+            cli = struct()
+            _, config = cli.parse()
+            for key in config.get_config().keys():
+                cli_option_set.add(key)
+
+        self.assertEqual(cli_option_set, all_tested_options_set)
+
+    def _test_boolean_option(self, option_struct):
+        option = option_struct.long_flag
+        option_with_underscores = self._convert_flag(option)
+        cli = option_struct.cli_subcommand()
+        _, config = cli.parse()
+        option_value = config.get_config().get(option_with_underscores).value()
+        self.assertEqual(option_value, False)
+
+        # Test boolean option
+        cli = option_struct.cli_subcommand()
+        cli.args.extend([option])
+        _, config = cli.parse()
+        option_value = config.get_config().get(option_with_underscores).value()
+        self.assertEqual(option_value, True)
+
+        # Test boolean option followed by value fails
+        cli = option_struct.cli_subcommand()
+        cli.args.extend([option, 'SHOULD_FAIL'])
+        with self.assertRaises(SystemExit):
+            _, config = cli.parse()
+
+    def _test_numeric_option(self, option_struct):
+        long_option = option_struct.long_flag
+        short_option = option_struct.short_flag
+        expected_value_string = option_struct.expected_value
+        expected_value = self._convert_string_to_numeric(
+            option_struct.expected_value)
+        default_value = None if option_struct.default_value == None else self._convert_string_to_numeric(
+            option_struct.default_value)
+
+        long_option_with_underscores = self._convert_flag(long_option)
+
+        # Test long_flag
+        cli = option_struct.cli_subcommand()
+        cli.args.extend([long_option, expected_value_string])
+        _, config = cli.parse()
+        option_value = config.get_config().get(
+            long_option_with_underscores).value()
+        self.assertEqual(option_value, expected_value)
+
+        # Test short_flag
```

Comment:
Is there any difference between all the test_short_flag in the various options? If not, seems like all of these should go into helper functions and be shared

Reply:
and the other ones too. Should be a bit cleaner now.